eventstore/test_tasks.py

```diff
@@ -703,3 +704,188 @@ def test_valid(self):
+
+
+class PostRandomContactsToSlackTests(TestCase):
+    def setUp(self):
+        tasks.rapidpro = TembaClient("textit.in", "test-token")
+
+    @responses.activate
+    @override_settings(
+        TURN_URL="https://turn",
+        TURN_TOKEN="token",
+        EXTERNAL_REGISTRATIONS_V2=True,
+        SLACK_CHANNEL="test-slack",
+        SLACK_URL="http://slack.com",
+        RAPIDPRO_URL="rapidpro",
+        RAPIDPRO_TOKEN="rapidpro-token",
+    )
+    def test_post_random_contacts_to_slack_channel(self):
+        responses.add(
+            responses.GET,
+            "https://textit.in/api/v2/contacts.json",
+            json={
+                "results": [
+                    {
+                        "uuid": "148947f5-a3b6-4b6b-9e9b-25058b1b7800",
+                        "name": "",
+                        "language": "eng",
+                        "groups": [],
+                        "fields": "",
+                        "blocked": False,
+                        "stopped": False,
+                        "created_on": "2015-11-11T08:30:24.922024+00:00",
+                        "modified_on": "2015-11-11T08:30:25.525936+00:00",
+                        "urns": ["whatsapp:27712345682"],
+                    },
+                    {
+                        "uuid": "128947f5-a3b6-4b3b-9e9b-25058b1b7801",
+                        "name": "",
+                        "language": "zul",
+                        "groups": [],
+                        "fields": "",
+                        "blocked": False,
+                        "stopped": False,
+                        "created_on": "2020-11-11T08:30:24.922024+00:00",
+                        "modified_on": "2021-11-11T08:30:25.525936+00:00",
+                        "urns": ["whatsapp:27720001010", "tel:0102584697"],
+                    },
+                ],
+                "next": None,
+            },
+        )
+
+        responses.add(
+            responses.GET,
+            "https://turn/v1/contacts/27712345682/messages",
+            json={
+                "chat": {
+                    "permalink": "https://turn.io/c/8cc14-6a4e-4f2-82ed-c5",
+                    "state_reason": "Re-opened by inbound message.",
+                    "unread_count": 0,
+                    "uuid": "68cc14b3-6a4e-4962-82ed-c572c6836fdd",
+                }
+            },
+        )
+
+        responses.add(
+            responses.GET,
+            "https://turn/v1/contacts/27720001010/messages",
+            json={
+                "chat": {
+                    "permalink": "https://turn.io/c/68c4-6ae-492-8ed-c6f",
+                    "state_reason": "Re-opened by inbound message.",
+                    "unread_count": 0,
+                    "uuid": "68h4b3-6a4e-4962-8ed-c572c36fdd",
+                }
+            },
+        )
+
+        responses.add(
+            responses.POST,
+            "http://slack.com/api/chat.postMessage",
+            json={
```

Comment:
This json is what the http request will return, not what it expects.

You should rather check the request after running the task to check the request body and url.

Example: https://github.com/praekeltfoundation/ndoh-hub/blob/develop/eventstore/test_tasks.py#L97-L114